aw_query/functions.py

```diff
@@ -47,31 +48,58 @@ def _verify_variable_is_type(variable, t):
 """
     Declarations
 """
-query2_functions = {}  # type: Dict[str, TQueryFunction]
+functions = {}  # type: Dict[str, TQueryFunction]
 
 
-def q2_function(f):
+def q2_function(transform_func=None):
     """
     Decorator used to register query functions.
 
     Automatically adds mock arguments for Datastore and TNamespace
     if not in function signature.
     """
+
+    def h(f):
+        sig = signature(f)
+        # If function lacks docstring, use docstring from underlying function in aw_transform
+        if transform_func and transform_func.__doc__ and not f.__doc__:
+            f.__doc__ = ".. note:: Documentation automatically copied from underlying function `aw_transform.{func_name}`\n\n{func_doc}".format(func_name=transform_func.__name__, func_doc=transform_func.__doc__)
+
+        @wraps(f)
+        def g(datastore: Datastore, namespace: TNamespace, *args, **kwargs):
```

Comment:
I really don't like this stuff, changing the arguments of a function on the fly is one ugly hack...
It was there before though so I guess it can stay.

Reply:
It's not a hack, it's just sugar. You might think it's ugly, but I think it's pretty clean and simple.

It was worse before, when all functions needed the datastore and namespace explicitly set even if they were never used (which throws linter warnings, for good reason).